File: backend/csharp-aspnet-webapi/Features/STT/STTFactory.cs
public class STTFactory : ISTTFactory
{
    private readonly IServiceProvider serviceProvider;

    public STTFactory(
        IServiceProvider serviceProvider
        )
    {
        this.serviceProvider = serviceProvider;
    }


    public ISTT GetSTT(UserSetting setting)
    {
        return setting.SttProviderKey switch
        {
            "WHISPER_CPP" =>
                serviceProvider.GetRequiredService<WhisperCpp>(),

            "PARAKEET" =>
                serviceProvider.GetRequiredService<Parakeet>(),

            _ =>
                throw new Exception("Unknown STT")
        };
    }
}

File: backend/java-springboot/src/main/java/com/lulo97/backend/features/stt/SttFactory.java
package com.lulo97.backend.features.stt;

import org.springframework.stereotype.Component;
import com.lulo97.backend.Utils;
import com.lulo97.backend.features.usersetting.UserSetting;

@Component
public class SttFactory {
    private final Whisper whisper;
    private final Parakeet parakeet;

    public SttFactory(Whisper whisper, Parakeet parakeet) {
        this.whisper = whisper;
        this.parakeet = parakeet;
    }

    public ISTT getSTT(UserSetting userSetting) {
        if (userSetting.getSttProviderKey() == Utils.WHISPER_CPP) {
            return whisper;
        } else {
            return parakeet;
        }
    }
}
